### PR TITLE
Prepare release 0.45.2

### PR DESCRIPTION
**Pull request by alekkemeny**
### Summary

This commit prepares the 0.45.2 release, this consists of bumping the version number to 0.45.2 in all the appropriate files. It also adds a prelude release note to provide a brief introduction to the release. When this commit merges it should be tagged as the 0.45.2 release.

### Details and comments

TODO:

- [ ]  Go through and make any updates to 0.45.2 release notes

### CODE REVIEW

**Review by 1ucian0 — DISMISSED** [review]
LGTM!

File: releasenotes/notes/fix-optimize-swap-before-measure-67e8896da2215d49.yaml [hunk 1]
```diff
@@ -3,8 +3,8 @@ fixes:
   - |
     Fixed an issue with the :class:`~.OptimizeSwapBeforeMeasure` pass where
     it would incorrectly optimize circuits involving swap and measure 
-    instructions. This commit fixes the bug by changing `DAGCircuit.successors`
-    to `DAGCircuit.descendants`. Also, added a couple of extra tests to ensure 
+    instructions. This commit fixes the bug by changing ``DAGCircuit.successors``
+    to ``DAGCircuit.descendants``. Also, added a couple of extra tests to ensure
```
Comment (ElePT) [inline releasenotes/notes/fix-optimize-swap-before-measure-67e8896da2215d49.yaml]: 2. This is me being nitpicky but it's always nicer to se API docs links (in this case, links for methods):
```suggestion
    instructions. This commit fixes the bug by changing :meth:`.DAGCircuit.successors`
    to :meth:`.DAGCircuit.descendants`. Also, added a couple of extra tests to ensure 
```

**Review by ElePT — COMMENTED** [review]
Thanks a lot, the release preparation part LGTM too, I just have two small comments on the release notes. 

1. There is a small typo on the blueprint circuit reno where there should be a backtick:

Screenshot 2024-01-18 at 13 18 07

Once these are addressed I think we can get started with the release :)

File: releasenotes/notes/fix-blueprintcircuit-phase-7102043cf2e47e33.yaml [hunk 2]
```diff
@@ -2,5 +2,5 @@
 fixes:
   - |
     Calling :meth:`~.QuantumCircuit.copy` or :meth:`~.QuantumCircuit.copy_empty_like` on a
-    :class:¬.BlueprintCircuit` will now correctly propagate the
+    :class:`¬.BlueprintCircuit` will now correctly propagate the
```
Comment (ElePT) [inline releasenotes/notes/fix-blueprintcircuit-phase-7102043cf2e47e33.yaml]:
```suggestion
    :class:`.BlueprintCircuit` will now correctly propagate the
```

**Review by ElePT — APPROVED** [review]
Thanks for the changes!! LGTM!!

**Pull request merged**